core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -85,37 +115,29 @@ final case class NonEmptyVector[A] private (vector: Vector[A]) {
    * universal .toString method.
    */
   def show(implicit A: Show[A]): String =
-    s"NonEmptyVector(${Show[Vector[A]].show(vector)})"
+    s"NonEmptyVector(${Show[Vector[A]].show(toVector)})"
 }
 
-private[data] sealed trait NonEmptyVectorInstances extends NonEmptyVectorLowPriority2 {
+private[data] sealed trait NonEmptyVectorInstances {
 
-  implicit def catsDataEqForNonEmptyVector[A](implicit A: Eq[A]): Eq[NonEmptyVector[A]] =
-    new Eq[NonEmptyVector[A]]{
-      def eqv(x: NonEmptyVector[A], y: NonEmptyVector[A]): Boolean = x === y
-    }
-
-  implicit def catsDataShowForNonEmptyVector[A](implicit A: Show[A]): Show[NonEmptyVector[A]] =
-    Show.show[NonEmptyVector[A]](_.show)
+  implicit val catsDataInstancesForNonEmptyVector: SemigroupK[NonEmptyVector] with Reducible[NonEmptyVector]
+      with Monad[NonEmptyVector] with Comonad[NonEmptyVector] with Traverse[NonEmptyVector] =
+    new NonEmptyReducible[NonEmptyVector, Vector] with SemigroupK[NonEmptyVector] with Monad[NonEmptyVector]
```

Comment:
I think this can be a `MonadRec` using the same style as:
https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/instances/list.scala#L31

and also I am pretty sure `MonadCombine`